internal/commands/app/create_test.go

```diff
@@ -230,222 +230,250 @@ func TestAppCreateHandler(t *testing.T) {
+				"  ---------------------"+dashBuffer+"  "+strings.Repeat("-", dirLength),
+				fmt.Sprintf("  Client App ID          "+spaceBuffer+fmtStr, "test-app-abcde"),
+				"  Realm Directory        "+spaceBuffer+localApp.RootDir,
+				fmt.Sprintf("  Realm UI               "+spaceBuffer+fmtStr, "http://localhost:8080/groups/123/apps/456/dashboard"),
+			)
+			if tc.cluster != "" {
+				display = append(display, fmt.Sprintf("  Data Source (Cluster)  "+spaceBuffer+fmtStr, "mongodb-atlas"))
+			}
+			if tc.dataLake != "" {
+				display = append(display, fmt.Sprintf("  Data Source (Data Lake)  "+fmtStr, "mongodb-datalake"))
+			}
+			display = append(display, "01:23:45 UTC DEBUG Check out your app: cd ./test-app && realm-cli app describe", "")
+			assert.Equal(t, strings.Join(display, "\n"), out.String())
+		})
+	}
+
+	testApp := realm.App{
+		ID:          "789",
+		GroupID:     "123",
+		ClientAppID: "from-app-abcde",
+		Name:        "from-app",
+	}
+
+	for _, tc := range []struct {
+		description    string
+		from           string
+		cluster        string
+		clients        cli.Clients
+		createDisplay  string
+		clusterDisplay string
+	}{
+		{
+			description:   "should create a minimal project dry run",
+			createDisplay: "minimal Realm app",
+		},
+		{
+			description: "should create a dry run for the specified from app",
+			from:        "from-app",
+			clients: cli.Clients{
+				Realm: mock.RealmClient{
+					FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+						return []realm.App{testApp}, nil
+					},
+				},
 			},
-		}, createdApp)
-
-		// TODO(REALMC-8262): Investigate file path display options
-		dirLength := len(localApp.RootDir)
-		fmtStr := fmt.Sprintf("%%-%ds", dirLength)
-
-		assert.Equal(t, strings.Join([]string{
-			"01:23:45 UTC INFO  Successfully created app",
-			fmt.Sprintf("  Info             "+fmtStr, "Details"),
-			"  ---------------  " + strings.Repeat("-", dirLength),
-			fmt.Sprintf("  Client App ID    "+fmtStr, "test-app-abcde"),
-			"  Realm Directory  " + localApp.RootDir,
-			fmt.Sprintf("  Realm UI         "+fmtStr, "http://localhost:8080/groups/123/apps/456/dashboard"),
-			fmt.Sprintf("  Data Source      "+fmtStr, "mongodb-atlas"),
-			"01:23:45 UTC DEBUG Check out your app: cd ./test-app && realm-cli app describe",
-			"",
-		}, "\n"), out.String())
-	})
-
-	t.Run("should create a minimal project dry run", func(t *testing.T) {
-		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
-		defer teardown()
-		profile.SetRealmBaseURL("http://localhost:8080")
-
-		out, ui := mock.NewUI()
-
-		cmd := &CommandCreate{
-			inputs: createInputs{
-				newAppInputs: newAppInputs{
-					Name:            "test-app",
-					Project:         "123",
-					Location:        realm.LocationVirginia,
-					DeploymentModel: realm.DeploymentModelGlobal,
+			createDisplay: "Realm app based on the Realm app from-app",
+		},
+		{
+			description: "should create a minimal project dry run with cluster set",
```

Comment:
[question] do we (care to) test the case where `dataLake` is set here?

Reply:
I don't think it's needed in this particular case, since it is so similar to the dry run cluster case and creates the need for the conditional spacing (if using a test table), which has been muddying the tests.

Reply:
i hear ya, but id prefer we have muddied waters here with more tests, than not.

can we break them out the differently-spaced tests into their own test funcs here (and live with copy/pasting that fixed-width logic for now)?  if/when we ever tackle that output to get it more deterministic, then we will see a lot of these test cases converge and we can re-table them at that point

Reply:
I was wrong here, the dry run tests are much easier to table